### ISSUE TITLE
[Bug]: Crash at ModelStore.persist due to NullPointerException

### ISSUE THREAD

**Issue by fanwgwg** [OneSignal/OneSignal-Android-SDK#1851]
### What happened?

```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'org.json.JSONObject com.onesignal.common.modeling.j.toJSON()' on a null object reference
       at com.onesignal.common.modeling.ModelStore.persist(ModelStore.kt:141)
       at com.onesignal.common.modeling.ModelStore.addItem(ModelStore.kt:108)
       at com.onesignal.common.modeling.ModelStore.addItem$default(ModelStore.kt:98)
       at com.onesignal.common.modeling.ModelStore.add(ModelStore.kt:46)
       at com.onesignal.common.modeling.IModelStore$DefaultImpls.add$default(IModelStore.kt:37)
       at com.onesignal.user.internal.subscriptions.impl.SubscriptionManager.addSubscriptionToModels(SubscriptionManager.kt:101)
       at com.onesignal.user.internal.subscriptions.impl.SubscriptionManager.addOrUpdatePushSubscription(SubscriptionManager.kt:63)
       at com.onesignal.notifications.internal.listeners.DeviceRegistrationListener$retrievePushTokenAndUpdateSubscription$1.invokeSuspend(DeviceRegistrationListener.kt:74)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:33)
       at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:102)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
       at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
       at kotlinx.coroutines.BlockingCoroutine.joinBlocking(BlockingCoroutine.java:85)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(BuildersKt__Builders.kt:59)
       at kotlinx.coroutines.BuildersKt.runBlocking(Builders.kt:1)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(BuildersKt__Builders.kt:38)
       at kotlinx.coroutines.BuildersKt.runBlocking$default(Builders.kt:1)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:60)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:59)
       at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Threads.kt:30)
```

### Steps to reproduce?

```Markdown
This crash is reported by crashlytics
```

### What did you expect to happen?

App should not crash

### OneSignal Android SDK version

5.0.1

### Android version

13

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jennantilla** [comment]
@fanwgwg thank you for reporting. We're investigating this and will let you know when a fix is ready.

**Comment by jkasten2** [comment]
We have made a number of improvements since the last report of this issue, can you try updating to the latest (5.1.8 at the time of this writing) and let us know if you can still reproduce this issue?

**Comment by jkasten2** [comment]
Based on the stack trace, this issue should have been fixed in [5.0.4](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/5.0.4), in PR #1876. However the newest version is always recommended.
